src/main/java/com/ocient/cli/extract/RecordExtractorFactory.java

```diff
@@ -81,8 +83,13 @@ CsvWriter create(int fileIndex) throws IOException{
     }
 
     // Simple utility function to help resolve the file name.
-    private String resolveFileName(int fileIndex){
-        String newFileName = extractConfig.getFilePrefix() + String.valueOf(fileIndex) + extractConfig.getFileExtension();
+    private String resolveFileName(int fileIndex, int threadNumber){
+        String newFileName = null;
+        if(isMultiThreaded){
+            newFileName = extractConfig.getFilePrefix() + String.valueOf(threadNumber) + "_" + String.valueOf(fileIndex) + extractConfig.getFileExtension();
+        } else {
+            newFileName = extractConfig.getFilePrefix() + String.valueOf(fileIndex) + extractConfig.getFileExtension();
+        }
```

Comment:
might be more readable?

```suggestion
        String newFileName = extractConfig.getFilePrefix();
        if (isMultiThreaded) {
            newFileName += String.valueOf(threadNumber) + "_";
        }
        newFileName += String.valueOf(fileIndex) + extractConfig.getFileExtension();
```

Reply:
changed